src/zcbor_encode.c

```diff
@@ -14,6 +14,17 @@
 _Static_assert((sizeof(size_t) == sizeof(void *)),
 	"This code needs size_t to be the same length as pointers.");
 
+
+#define FAIL_IF(expr, err) \
+do {\
+	if (expr) { \
+		ZCBOR_ERR(err); \
+	} \
+} while(0)
+
+#define CHECK_PAYLOAD() FAIL_IF(state->payload >= state->payload_end, ZCBOR_ERR_NO_PAYLOAD)
```

Comment:
Are these duplicates of what is already in `zcbor_decode.c`? If so: should we pull them to a common .h file to avoid regression when one is changed but not the other.

Reply:
Yep. Moved to zcbor_common.h